File: my-submissions/m606.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def tree2str(self, root: Optional[TreeNode]) -> str:
        if not root :
            return ''
        if not root.left and not root.right :
            return str(root.val)

        output = [str(root.val)]
        if root.left :
            output.extend(['(', self.tree2str(root.left), ')'])
        else :
            output.append('()')
        if root.right :
            output.extend(['(', self.tree2str(root.right), ')'])

        return ''.join(output)

File: my-submissions/m606.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public String tree2str(TreeNode root) {
        if (root == null)
            return "";
        if (root.left == null && root.right == null)
            return "" + root.val;

        return root.val
                + "(" + tree2str(root.left) + ")"
                + (root.right != null ? "(" + tree2str(root.right) + ")" : ""); 
    }
}